JOSESwift/Sources/AESCBCEncryption.swift

```diff
@@ -63,46 +69,69 @@ struct AESCBCEncryption {
         additionalAuthenticatedData: Data,
         authenticationTag: Data
     ) throws -> Data {
-        // Check if the key length contains both HMAC key and the actual symmetric key.
-        guard contentEncryptionAlgorithm.checkKeyLength(for: contentEncryptionKey) else {
-            throw JWEError.keyLengthNotSatisfied
-        }
-
-        // Get the two keys for the HMAC and the symmetric encryption.
-        let keys = try contentEncryptionAlgorithm.retrieveKeys(from: contentEncryptionKey)
-        let hmacKey = keys.hmacKey
-        let decryptionKey = keys.encryptionKey
-
         // Put together the input data for the HMAC. It consists of A || IV || E || AL.
         var concatData = additionalAuthenticatedData
         concatData.append(initializationVector)
         concatData.append(ciphertext)
         concatData.append(additionalAuthenticatedData.getByteLengthAsOctetHexData())
 
-        // Calculate the HMAC for the concatenated input data and compare it with the reference authentication tag.
         let hmacOutput = try HMAC.calculate(
             from: concatData,
             with: hmacKey,
-            using: contentEncryptionAlgorithm.hmacAlgorithm
+            using: hmacAlgorithm
         )
 
         guard
-            authenticationTag.timingSafeCompare(with: contentEncryptionAlgorithm.authenticationTag(for: hmacOutput))
+            authenticationTag.timingSafeCompare(with: try getAuthenticationTag(for: hmacOutput))
         else {
             throw JWEError.hmacNotAuthenticated
         }
 
         // Decrypt the cipher text with a symmetric decryption key, a symmetric algorithm and the initialization vector,
-        // return the plaintext if no error occured.
+        // return the plaintext if no error occurred.
         let plaintext = try AES.decrypt(
             cipherText: ciphertext,
-            with: decryptionKey,
+            with: encryptionKey,
             using: contentEncryptionAlgorithm,
             and: initializationVector
         )
 
         return plaintext
     }
+
+    static func getHMACAlgorithm(for contentEncryptionAlgorithm: ContentEncryptionAlgorithm) throws -> HMACAlgorithm {
+            switch contentEncryptionAlgorithm {
+            case .A256CBCHS512:
+                return .SHA512
+            case .A128CBCHS256:
+                return .SHA256
+            case .A256GCM, .A128GCM:
+                throw JWEError.contentEncryptionAlgorithmMismatch
+            }
+    }
+
+    static func retrieveKeys(for contentEncryptionAlgorithm: ContentEncryptionAlgorithm,
+                             from inputKey: Data) throws -> (hmacKey: Data, encryptionKey: Data) {
```

Comment:
```suggestion
    static func retrieveKeys(
        for contentEncryptionAlgorithm: ContentEncryptionAlgorithm,
        from inputKey: Data
    ) throws -> (hmacKey: Data, encryptionKey: Data) {
```

Reply:
IIRC this is the formatting we were using in other parts of the codebase

Reply:
😅 I wonder if I can install a formatter to match styles automatically.
Is there something you would recommend? 
We could go with https://github.com/apple/swift-format

Reply:
Yeah we should definitely automate this. IIRC we did some research in the past but I'm sure things changed in the meantime. Defaulting to Apple's formatter surely isn't a bad idea. Probably what Xcode uses under the hood too (I guess) so that might make it easier as well.